### ISSUE TITLE
Duplicating page links embedded menu element

### ISSUE THREAD

**Issue by shchukax** [microweber/microweber#343]
Title may not be easy to understand, but I'll try to describe.  I have two custom menus set up, each with a set of links to specific pages.  I then have a page from the 'text page' template, which contains three columns.  In the right column, I inserted one of the custom menus.  It gives me what I need.

Now I need a similar page, with three columns - the content is different and it should use a different custom menu, but everything else layout-wise is the same.  I used 'duplicate' button on page settings page and created the second page. I then edit this second page, delete the custom menu (it's from the first page) and add the other custom menu.

Now, when I went back to the first page, the custom menu on it was changed to the second custom menu.  Now, whenever I change the menu, it automatically changes on both pages.  The only link between the two pages is that the second was created as a duplicate of the first.  They should _not_ be linked for element editing purposes.

**Comment by peter-mw** [comment]
hi @shchukax this happens because the page get duplicated and the module ID is preserved in the content saved in the DB

to solve the problem you must delete the module and add it again in order to obtain new module ID

**Comment by shchukax** [comment]
hi @peter-mw Yes, this is exactly what I did to resolve it.  However "duplicate" is rather misleading as the function name.  It doesn't duplicate the page, it only duplicates the main content - but all modules are still linked.  I would assume that "duplicate" means create a copy of everything on the page.  Either way, as there's a simple workaround, this isn't the highest priority.

**Comment by peter-mw** [comment]
yes i agree 

probably i must make a way to change the module's id from the settings

**Comment by shchukax** [comment]
Hm.. A non-technical user probably wouldn't know what to do with that ID; they would probably expect the internal "duplication" process to automatically create another module in the module table and put the new ID into the content table for the newly created page.

**Comment by peter-mw** [comment]
so it must create copy of each module and also copy its settings from the original module

**Comment by shchukax** [comment]
Yeah... I see how it can get complicated very quickly.  Maybe it would better to just document the duplication process better - the workaround is simple, but I didn't realise what was happening until you replied to my comment.

**Comment by peter-mw** [comment]
yes is still not resolved, and i am still thinking how to make it

its a bit tricky as it need to get all modules in the content and copy their settings 

the copy function is here https://github.com/microweber/microweber/blob/1.0.8/src/Microweber/Providers/Content/ContentManagerHelpers.php#L420 


but needs more work to resolve this issue
